addons/account/wizard/account_advance_payment.py

```diff
@@ -0,0 +1,338 @@
+            amount = self.default_get(['amount']).get('amount')
+            return {'value': {'amount': amount}}
+
+    #=== CONSTRAINT METHODS ===#
+
+    def _check_amount_is_positive(self):
+        for wizard in self:
+            if (wizard.advance_payment_method == 'percentage' and wizard.amount <= 0.00) \
+                    or (wizard.advance_payment_method == 'fixed' and wizard.fixed_amount <= 0.00):
+                raise UserError(_('The value of the down payment amount must be positive.'))
+
+    # === ACTION METHODS ===#
+
+    def create_invoices(self):
+        self._check_amount_is_positive()
+        invoices = self._create_invoices(self.order_ids)
+        return self.order_ids.action_view_invoice(invoices=invoices)
+
+    def view_draft_invoices(self):
+        return {
+            'name': _('Draft Invoices'),
+            'type': 'ir.actions.act_window',
+            'view_mode': 'tree',
+            'views': [(False, 'list'), (False, 'form')],
+            'res_model': 'account.move',
+            'domain': [('state', '=', 'draft')], # To override
+        }
+
+    def _create_invoices(self, orders):
+        if self.advance_payment_method == 'delivered':
+            return orders._create_invoices(final=True, grouped=not self._invoice_grouped())
+
+        return self.create_down_payment_invoice()
+
+    def create_down_payment_invoice(self):
+        self.order_ids.ensure_one()
+        self = self.with_company(self.company_id)
+        order = self.order_ids
+
+        # Create down payment section if necessary
+        OrderLine = self.order_ids._subrecord_type().with_context(no_log_for_new_lines=True)
+        if not any(line.display_type and line.is_downpayment for line in order.order_line):
+            OrderLine.create(
+                self._prepare_down_payment_section_values(order)
+            )
+
+        values, accounts = self._prepare_down_payment_lines_values(order)
+        down_payment_lines = OrderLine.create(values)
+
+        invoice = self.env['account.move'].sudo().create(
+            self._prepare_invoice_values(order, down_payment_lines, accounts)
+        ).with_user(self.env.uid)  # Unsudo the invoice after creation
+
+        # Ensure the invoice total is exactly the expected fixed amount.
+        if self.advance_payment_method == 'fixed':
+            delta_amount = (invoice.amount_total - self.fixed_amount) * (-1 if invoice.is_inbound() else 1) * self.order_ids._sign()
+            if not order.currency_id.is_zero(delta_amount):
+                receivable_line = invoice.line_ids \
+                                      .filtered(lambda aml: aml.account_id.account_type == self._account_type())[:1]
+                product_lines = invoice.line_ids \
+                    .filtered(lambda aml: aml.display_type == 'product')
+                tax_lines = invoice.line_ids \
+                    .filtered(lambda aml: aml.tax_line_id.amount_type not in (False, 'fixed'))
```

Comment:
Peculiar formatting, I guess IDE reformatted the pasting ?

Reply:
Nope, came straight from the Sales code ;) But I put it back on one line.